kotlin issue and compatibility fix (#2256)
segmentio/segment-docs: src/connections/sources/catalog/libraries/mobile/kotlin-android/index.md

    2. Search for **Kotlin (Android)** and click **Add source**.
2. Add the Analytics dependency to your build.gradle.

>     Segment recommends you to install the library with a build system like Gradle, as it simplifies the process of upgrading versions and adding integrations. The library is distributed through [Maven Central](https://repo1.maven.org/maven2/com/segment/analytics/kotlin/android/){:target="_blank"}. Add the analytics module to your build.gradle as a dependency as shown in the code sample below, and replace `<latest_version>` with the latest version listed in our [releases page](https://github.com/segmentio/analytics-kotlin/releases){:target="_blank"}

Reviewer: ```suggestion
    Segment recommends you to install the library with a build system like Gradle, as it simplifies the process of upgrading versions and adding integrations. The library is distributed through [Maven Central](https://repo1.maven.org/maven2/com/segment/analytics/kotlin/android/){:target="_blank"}. Add the analytics module to your build.gradle as a dependency as shown in the code sample below, and replace `<latest_version>` with the latest version listed on Segment's [releases page](https://github.com/segmentio/analytics-kotlin/releases){:target="_blank"}
```

      }
    ```

>     > warning ""
>     > **Note:** In android, application context is required to pass as the second parameter, or an error `Using JVM Analytics initializer in Android platform. Context is required in constructor!` will be thrown out.
> 

Reviewer: ```suggestion
    **Note:** If you're on an Android platform, you must add the application context as the second parameter.
```

Reviewer: we can't add the warning label format because it messes with the numbering format

>     Automatically tracking lifecycle events (`Application Opened`, `Application Installed`, `Application Updated`) is optional, but Segment highly recommends you to configure these options in order to track core events.

Reviewer: ```suggestion
    <br>Automatically tracking lifecycle events (`Application Opened`, `Application Installed`, `Application Updated`) is optional, but Segment highly recommends you to configure these options in order to track core events.
```

      }
    ```

>     **Note:** If you're on an Android platform, you must add the application context as the second parameter.

Reviewer: ```suggestion
    **Note: If you're on an Android platform, you must add the application context as the second parameter.**
```

    <!-- Required for internet. -->
    <uses-permission android:name="android.permission.INTERNET"/>
    <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE"/>
    ```

> 5. Enable Java 8+ API desugaring
>    
>    The SDK internally uses a number of Java 8 language APIs through desugaring. Please make sure your project:
>    * either enables desugaring (see how to enable it [here](https://developer.android.com/studio/write/java8-support#library-desugaring))
>    * or requires a minimum API level of 26.

Reviewer: ```suggestion
5. Enable Java 8+ API desugaring.
   
    The SDK internally uses a number of Java 8 language APIs through desugaring. Make sure your project either [enables desugaring](https://developer.android.com/studio/write/java8-support#library-desugaring)) or requires a minimum API level of 26.
```

Segment recommends you to test your destination implementation end-to-end. Send some sample analytics events and ensure that they reach the destination.

## Compatibility

> [View the Analytics-Kotlin compatibility notes on GitHub](https://github.com/segmentio/analytics-kotlin#compatibility).

Reviewer: ```suggestion
If you use a pure Java codebase, please refer to [Java Compatibility](https://github.com/segmentio/analytics-kotlin/blob/main/JAVA_COMPAT.md) for sample uses.
```

> If you use a pure Java codebase, please refer to [Java Compatibility](https://github.com/segmentio/analytics-kotlin/blob/main/JAVA_COMPAT.md){:target="_blank"} for sample uses.

Reviewer: ```suggestion
If you use a Java codebase, please refer to [Java Compatibility docs](https://github.com/segmentio/analytics-kotlin/blob/main/JAVA_COMPAT.md){:target="_blank"} for sample uses.
```

Segment recommends you to test your destination implementation end-to-end. Send some sample analytics events and ensure that they reach the destination.

## Compatibility

> [View the Analytics-Kotlin compatibility notes on GitHub](https://github.com/segmentio/analytics-kotlin#compatibility).

Reviewer: ```suggestion
If you use a pure Java codebase, please refer to [Java Compatibility](https://github.com/segmentio/analytics-kotlin/blob/main/JAVA_COMPAT.md) for sample uses.
```

Reviewer: @wenxi-zeng @prayansh the "Compatibility" section you linked to is pretty short on GitHub, so I thought we could just include what's written there directly onto the docs site. However, I didn't include the bit about Java 8 language API desugaring because that information is already included in Step 5 of this doc. What do you think?

Author: yeah, I think this is good. the reason I linked the whole `compatibility` section is just in case in the future we are adding more compatibility stuff. but for now, we only have two. and I think we will not have anything extra in the near future. we can update this section when that does happen
